rclcpp/include/rclcpp/node.hpp

```diff
@@ -936,17 +938,69 @@ class Node : public std::enable_shared_from_this<Node>
    * of the latter things,
    * rclcpp::exceptions::ParameterModifiedInCallbackException will be thrown.
    *
-   * There may only be one callback set at a time, so the previously set
-   * callback is returned when this method is used, or nullptr will be returned
-   * if no callback was previously set.
+   * The callback functions must remain valid as long as the
+   * returned unique pointer is valid.
+   * The returned unique pointer can be promoted to a shared version.
+   *
+   * Releasing the returned unique pointer unregisters the callback.
+   * `remove_on_set_parameters_callback` can also be used.
+   *
+   * All the registered callbacks are called when a parameter is set.
+   * The order of the callbacks is not important.
+   *
+   * \param callback The callback to register.
+   * \returns A unique pointer. The callback is valid as long as the unique pointer is alive.
+   * \throws std::bad_alloc if the allocation of the OnSetParametersCallbackHandle fails.
+   */
+  RCLCPP_PUBLIC
+  OnSetParametersCallbackHandle::UniquePtr
+  add_on_set_parameters_callback(OnParametersSetCallbackType callback);
+
+  /// Remove a callback registered with `add_on_set_parameters_callback`.
+  /**
+   * Releases a handler, returned by `add_on_set_parameters_callback`.
+   *
+   * e.g.:
+   *
+   *    `remove_on_set_parameters_callback(scoped_callback.get())`
+   *
+   * As an alternative, the unique pointer can be released:
+   *
+   *    `scoped_callback.reset()`
+   *
+   * Supposing that `scoped_callback` was the only owner.
+   *
+   * Calling `remove_on_set_parameters_callback` more than once with the same handler,
+   * or calling it after the shared pointer has been released is an error.
+   * Releasing the smart pointer after calling `remove_on_set_parameters_callback` is not a problem.
```

Comment:
Releasing != reset, see:

https://en.cppreference.com/w/cpp/memory/unique_ptr/release vs https://en.cppreference.com/w/cpp/memory/unique_ptr/reset, release will not destruct it. We should use consistent terminology here and above.

Reply:
I'd say, instead, "Resetting or letting the smart pointer go out of scope...".

Reply:
Yes thanks, I sometimes mix those words.